lib/spack/spack/build_systems/intel.py

```diff
@@ -103,6 +135,711 @@ def _filtered_components(self):
+                d = Prefix(ancestor(d)).bin.intel64
+                # NB: psxevars.sh is wrested specially from studio_dir, in
+                # intel-parallel-studio/package.py:file_to_source
+            else:
+                d = d.bin
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    def component_lib_dir(self, component=None, relative=False):
+        # Provide starting directory for find_libraries() and
+        # SPACK_COMPILER_EXTRA_RPATHS.
+        d = self.component_dir(component)
+
+        if sys.platform == 'darwin':
+            d = d.lib
+        else:
+            if component == 'mpi' or 'intel-mpi' in self.name:
+                d = d.intel64.lib
+            else:
+                d = d.lib.intel64
+            # A bit weird, but  I'm sure there are good reasons for it.
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    @property
+    def component_include_dir(self, component=None, relative=False):
+        d = self.component_dir(component)
+
+        if component == 'mpi' or 'intel-mpi' in self.name:
+            d = d.intel64
+        d = d.include
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    @property
+    def omp_libs(self):
+        # Supply LibraryList for linking OpenMP
+
+        # FIXME  if sys.platform == 'darwin' ....
+
+        # TODO?  Across Spack packages for "client" applications, are variants
+        #   named consistently enough to determine if OpenMP is even needed?
+
+        if '%intel' in self.spec:
+            omp_libnames = ['libiomp5']
+
+            # Note about search root: For MKL, the directory
+            # "$MKLROOT/../compiler" will be present even for an MKL-only
+            # product installation (as opposed to one being ghosted via
+            # packages.yaml), specificially to provide the 'iomp5' libs.
+
+            omp_libs = find_libraries(
+                omp_libnames,
+                root=self.component_lib_dir(component='compiler'),
+                shared=self._want_shared)
+
+        elif '%gcc' in self.spec:
+            gcc = Executable(self.compiler.cc)
+            omp_libnames = gcc('--print-file-name',
+                               'libgomp.{0}'.format(dso_suffix),
+                               output=str)
+            omp_libs = LibraryList(omp_libnames)
+
+        if len(omp_libs) < 1:
+            raise_lib_error('Cannot locate OpenMP libraries:', omp_libnames)
+
+        debug_print(omp_libs)
+        return omp_libs
+
+    @property
+    def blas_libs(self):
+        # Main magic here.
+        # For reference, see The Intel Math Kernel Library Link Line Advisor:
+        # https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor/
+
+        mkl_libnames = ['libmkl_intel_' + self.intel64_int_suffix]
+        if not self.spec.satisfies('threads=openmp'):
```

Comment:
```
if self.spec.satisfies('threads=none'):
  ...
elif self.spec.satisfies('threads=openmp'):
  if '%intel' in self.spec:
     ...
  elif ...
     ...
```

that's because eventually one could add TBB here as well.

IMO this also reads better, of course the result is the same as you currently have.